### ISSUE TITLE
Add PathValidator support for YenKShortestPath algorithm

### ISSUE THREAD

**Issue by ajayslele** [jgrapht/jgrapht#827]
```
 * JGraphT version: 1.3.1
 * Java version (java -version)/platform:  Java 8/Linux
```

**Issue**
YenKShortestPath algorithm does not support path validation while the path is getting built, like how it is supported on KShortestSimplePaths (ref. https://jgrapht.org/javadoc/org/jgrapht/alg/shortestpath/KShortestSimplePaths.html#KShortestSimplePaths-org.jgrapht.Graph-int-org.jgrapht.alg.shortestpath.PathValidator-). This support is very useful when paths only matching certain criteria are required. Opening this ticket to request support for that

**Steps to reproduce (small coding example)**

**Expected behaviour**

**Other information**

**Comment by SChudakov** [comment]
Hey, sorry for possibly late answer. I am the author of the `YenKShortestPath` implementation and I will look into if it is possible to implement this feature.

**Comment by lucacolli** [comment]
I too would need a lot this implementation! Thank you!

**Comment by Toptachamann** [comment]
Resolved in #891
